### PR TITLE
[Multi-Screen Window Placement API ] Update article for second OT

### PR DESCRIPTION
**Pull request by tomayac**
Updated the article for the second OT.

Do not merge before the OT URL is up.

### CODE REVIEW

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (quisquous) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:69]:
```suggestion
Starting in Chromium&nbsp;93, the Multi-Screen Window Placement API will again be available as an
origin trial in Chromium. This second origin trial is expected to end in Chromium&nbsp;96.
```

We had a discussion today and decided to push this back to M93.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (quisquous) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:199]: Is it worth mentioning that this prompts the user for permission here?

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (quisquous) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:299]: Thanks for the text on all the new events.  These look good to me!

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:199]: That is covered under "The window-placement permission", but this might be an okay place to remind readers of that.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:182]: Please update this link, maybe https://webscreens.github.io/window-placement/#usage-overview-screen-information:~:text=This%20method%20may%20prompt%20the%20user%20for%20permission

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:283]: `screenschange` does *not* fire when attributes of a particular screen change, only when some screens are added or removed from the `screens` array. For that, users should use the new `change` event on `Screen` (and `ScreenAdvanced`). Let us know if we can clarify https://webscreens.github.io/window-placement/#api-screens-events or the explainer, etc.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:288]: I closed that issue; we won't add data to the events themselves. Please let us know if you have remaining ergonomics issues/suggestions regarding events, thanks!

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:289]: They can, but really shouldn't need to call getScreens() again. Instead, they can/should use the *live* multi-screen data of a cached `Screens` interface object.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:294]: Consider avoiding the repeated call to getScreens. One practical example here might be something like:

const screensInterface = await window.getScreens();
let cachedScreensLength = screensInterface.screens.length;
screensInterface.addEventListener('screenschange', event => {
  if (screensInterface.screens.length != cachedScreensLength) {
    console.log('The screen count changed from' + cachedScreensLength + ' to' screensInterface.screens.length);
    cachedScreensLength = screensInterface.screens.length;
  }
});

Generally, `screenschange` should correspond to a change in the length of `screens`, but technically it's possible that a single event might fire if N screens were added and N screens were removed simultaneously...

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:307]: Similarly, reuse screensInterface instead of calling getScreens() again

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:320]: ditto

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:361]: nit: "that is, the screen change events and" (there are multiple events that wouldn't fire)

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:480]: Let's credit enne with API development too!

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:182]:
```suggestion
[can](https://webscreens.github.io/window-placement/#usage-overview-screen-information:~:text=This%20method%20may%20prompt%20the%20user%20for%20permission)
```

Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:182]: Done.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:199]:
```suggestion
second screen using `Window.getScreens()`. Calling this function will show a permission prompt that asks me whether the site may open and place windows on my screen. The function returns a promise that resolves with an array of
```

Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:199]: I'll add it.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:284]:
```suggestion
that "screens" is plural in the event name.) This means the event fires whenever a new or an existing screen is (physically or virtually in the
case of Sidecar) plugged in or unplugged.
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:283]: Thanks, clarified.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:289]:
```suggestion
itself does not provide this data. To look up the screen details, use the live object from a cached `Screens` interface.
```

Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:289]: Understood.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:296]:
```suggestion
const screensInterface = await window.getScreens();
let cachedScreensLength = screensInterface.screens.length;
screensInterface.addEventListener('screenschange', (event) => {
  if (screensInterface.screens.length !== cachedScreensLength) {
    console.log(`The screen count changed from ${cachedScreensLength} to ${screensInterface.screens.length}`);
    cachedScreensLength = screensInterface.screens.length;
  }
});
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:299]: :-) Thanks for making the events more explicit, this will help developers subscribe to just the things they need info on.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:309]:
```suggestion
const screensInterface = await window.getScreens();
screensInterface.addEventListener('currentscreenchange', async (event) => {
  const details = screensInterface.currentScreen;
  console.log('The current screen has changed.', event, details);
});
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:320]:
```suggestion
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:321]:
```suggestion
  console.log('The first screen has changed.', event, firstScreen);
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:361]: Gotcha.

Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:361]:
```suggestion
The other aspects of the API, that is, the various screen change events and the `screen` property of the
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:480]:
```suggestion
[Mike Wasserman](https://www.linkedin.com/in/mike-wasserman-9900a079/) and [Adrienne Walker](https://github.com/quisquous). This article was reviewed by
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:287]: With the live objects and the caching of the initial call result of `getScreens()` it still feels like the events passed to the event listeners are a bit superfluous, but it's definitely workable.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md]: Noted and changed.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md]: Done.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md]: Done.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md]: Done.

**Comment by tomayac** [comment]
Thanks, @quisquous and @michaelwasserman. I have addressed your remarks and comments :-) PTAL!

Not sure who, but either of you needs to register their GitHub account with Google (https://opensource.corp.google.com/github/) so the Google CLA bot is happy (it's [grumpy](https://github.com/GoogleChrome/web.dev/pull/5591#issuecomment-859447895) right now).  After that, please comment `@googlebot I fixed it.`.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [reply src/site/content/en/blog/multi-screen-window-placement/index.md]: Is it possible and acceptable API design to trigger event listeners without passing an event object? If so, maybe that would be desirable here?

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:397]: The feature now has permission policy support, so if an iframe will actually attempt to use the API, it will the iframe will need an "allow='window-placement'" attribute.

Comment (michaelwasserman) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:397]: Hmm, clicking elmer just opens the demo in a separate tab, so perhaps we don't need to support the iframe itself using getScreens(), etc.

**Comment by michaelwasserman** [comment]
Thanks Tom! The updated article lgtm, modulo the potential iframe permission policy issue.

Thanks for updating the demo too, but I think it might need a bit of additional polish: Clicking a popup on the primary/internal screen on Mac enters and immediately exits fullscreen, then closes and reopens all popups. I'm guessing screenschange / currentscreenchange / screen.change is fired when entering/exiting fullscreen. This could be because my mac's gpu switches (http://crbug.com/1218952) or just because the current screen metrics (e.g. the avail* dimensions) may rightly change on fullscreen toggle. Hopefully we can refine the demo a bit to avoid super glitchy cases; lmk if you'd like some help with that.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:397]: I have (had) logic in the demo that would recognize when it was iframed and open in a new tab. With the new permission policy integration we can get rid of this.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:399]:
```suggestion
    allow="window-placement" sandbox="allow-popups allow-popups-to-escape-sandbox allow-scripts"
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:10]:
```suggestion
updated: 2021-06-14
```

Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:10]:
```suggestion
updated: 2021-06-14
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:399]:
```suggestion
    allow="fullscreen; window-placement" sandbox="allow-modals allow-popups allow-popups-to-escape-sandbox allow-scripts"
```

**Comment by tomayac** [comment]
Thanks, @michaelwasserman! I have now modified the listener for `currentscreenchange` to be a no-op (apart from logging), since it always fires whenever fullscreen is entered. I tried detecting this with `document.fullscreen`, but it wouldn't work. We can revisit this when http://crbug.com/1218952 gets fixed.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (jpmedley) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:10]: Do you know when the OT will be live? We could use `scheduled: true` and merge this as soon as the edit is done. (I prefer that because nobody has to remember to loop back and merge this.)

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (jpmedley) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:183]:
```suggestion
choose to show the permission prompt dynamically on the first attempt to use any of the methods of
```

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (jpmedley) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:271]: Isn' necessarily or doesn't have to be? The former implies an element of chance, while the later implies my complete control.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (jpmedley) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:276]:
```suggestion
The `currentScreen` field is a live object corresponding to the current `window.screen`. The object
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:10]: @michaelwasserman will know.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:271]: "Isn't necessarily". Operating systems may have different heuristics as to which plugged in additional screen they make the primary screen.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (michaelwasserman) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:10]: We are pretty confident we'll start OT2 in M93. Maybe set the date around the first M93 Beta (Thu July 29)?

Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md:10]: @michaelwasserman, if you ping the URL once you know it, I can then put it into the article and take care of merging if @jpmedley approves the article in the current state, with the only pending addition being the URL (and updating the publishing date to whatever date it may be).

Comment (tomayac) [reply src/site/content/en/blog/multi-screen-window-placement/index.md]: I have added the `updated` date that is in the far future :-) of July and the `scheduled: true` flag. Not sure if this works for already published articles, but with the `Do not merge` label we're also on the safe side.

**Review by jpmedley — APPROVED** [review]

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (quisquous) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:18]:
```suggestion
  url: https://developer.chrome.com/origintrials/#/view_trial/-8087339030850568191
```

Here's the new origin trial link.

File: src/site/content/en/blog/multi-screen-window-placement/index.md
Comment (tomayac) [inline src/site/content/en/blog/multi-screen-window-placement/index.md:50]:
```suggestion
| 4. **Origin trial**                          | **[In progress](https://developer.chrome.com/origintrials/#/view_trial/-8087339030850568191)**                 |
```

**Pull request merged**
